kratos/expression/variable_expression_data_io.cpp

```diff
@@ -230,6 +483,24 @@ void VariableExpressionDataIO<Matrix>::Read(
     }
 }
 
+template<>
+void VariableExpressionDataIO<std::vector<Matrix>>::Read(
+    RawLiteralFlatExpression& rExpression,
+    const IndexType EntityIndex,
+    const std::vector<Matrix>& Value) const
+{
+    const auto& r_shape = rExpression.GetItemShape();
+    const IndexType matrix_size = r_shape[1] * r_shape[2];
```

Comment:
make sure `r_shape` has the correct size

Reply:
as mentioned in the previous comment, added a full debug error and a comment.